### PR TITLE
The final physics update PR to RRFS.v1

### PR DESCRIPTION
**Pull request by haiqinli**
This is the final physics updates PR to RRFS.v1, and it includes some minor physics bug fixies of RUC LSM, FMS and smoke/dust.

### CODE REVIEW

File: physics/smoke_dust/module_add_emiss_burn.F90
Comment (grantfirl) [inline physics/smoke_dust/module_add_emiss_burn.F90:58]: You can edit out the comment to Jordan since this was done in https://github.com/NOAA-EMC/fv3atm/pull/841

File: physics/smoke_dust/rrfs_smoke_wrapper.meta
Comment (grantfirl) [inline physics/smoke_dust/rrfs_smoke_wrapper.meta]: units should be `1` according to https://github.com/ESCOMP/CCPPStandardNames/blob/main/StandardNamesRules.rst. This change should be here and everywhere else this variable is used in ccpp/physics and fv3atm.

File: physics/smoke_dust/rrfs_smoke_wrapper.meta
Comment (grantfirl) [inline physics/smoke_dust/rrfs_smoke_wrapper.meta]: units should be 1 according to https://github.com/ESCOMP/CCPPStandardNames/blob/main/StandardNamesRules.rst. This change should be here and everywhere else this variable is used in ccpp/physics and fv3atm.

**Review by grantfirl — CHANGES_REQUESTED** [review]
Minor changes requested to metadata and comments

File: physics/MP/Thompson/module_mp_thompson.F90
Comment (dustinswales) [inline physics/MP/Thompson/module_mp_thompson.F90]: Change to real(wp)

File: physics/PBL/MYNN_EDMF/module_bl_mynn.F90
Comment (dustinswales) [inline physics/PBL/MYNN_EDMF/module_bl_mynn.F90]: Change `real(kmaxfire)` to `real(kmaxfire, kind=kind_phys`)

File: physics/smoke_dust/rrfs_smoke_wrapper.F90
Comment (dustinswales) [inline physics/smoke_dust/rrfs_smoke_wrapper.F90]: Change `real(hour_int)` to `real(hour_int, type=kind_phys)`

**Review by dustinswales — CHANGES_REQUESTED** [review]
@haiqinli Thanks for making these changes. Just a few tiny requests.

**Comment by jordanschnell** [comment]
@haiqinli @dustinswales - can I make this change (above for precip_factor) directly or does Haiqin need to do it?

**Comment by grantfirl** [comment]
> @haiqinli @dustinswales - can I make this change (above for precip_factor) directly or does Haiqin need to do it?

It depends on whether you have push access to Haiqin's fork. If he adds your GitHub user as a collaborator on his fork of ccpp-physics, then you should have push permissions and can make commits.

**Comment by haiqinli** [comment]
@jordanschnell I am back, and I can update the PR to include your changes.

**Comment by haiqinli** [comment]
@grantfirl Some smoke code updates will come next Monday. Let's hold this PR a couple of days. Thanks.

File: physics/smoke_dust/rrfs_smoke_wrapper.F90
Comment (haiqinli) [reply physics/smoke_dust/rrfs_smoke_wrapper.F90]: @dustinswales I updated the PR with above. However, the compilation failed as below. Did I miss something? Thanks.

/scratch2/BMC/acomp/Haiqin.Li/Commit/RRFS.v1/ufs-weather-model/FV3/ccpp/physics/physics/smoke_dust/rrfs_smoke_wrapper.F90:954:41:

  954 |     precip_factor  = 2.5_kind_phys + real(hour_int, type=kind_phys)*5._kind_phys/24._kind_phys
      |                                         1
Error: Cannot find keyword named 'type' in call to 'real' at (1)

Comment (dustinswales) [reply physics/smoke_dust/rrfs_smoke_wrapper.F90]: Sorry, typo on my end. It should be "kind" not "type".

Comment (haiqinli) [reply physics/smoke_dust/rrfs_smoke_wrapper.F90]: @dustinswales Thanks. I will update accordingly.

**Review by grantfirl — APPROVED** [review]
Minor changes requested to metadata and comments

**Comment by grantfirl** [comment]
@haiqinli Do you still want to hold this, or should it be scheduled on the merge queue?

**Review by Qingfu-Liu — APPROVED** [review]

**Comment by haiqinli** [comment]
> @haiqinli Do you still want to hold this, or should it be scheduled on the merge queue?

@grantfirl I am updating this PR, and we can schedule it on the queue. Thanks.

**Comment by jkbk2004** [comment]
@grantfirl @dustinswales @MatthewPyle-NOAA Tests are done at https://github.com/ufs-community/ufs-weather-model/pull/2298. Can you merge this pr?

**Comment by MatthewPyle-NOAA** [comment]
@jkbk2004 Ability to merge looks to be held up by an old block.

**Review by dustinswales — APPROVED** [review]

**Pull request merged**
